components/kubelet/pod/component.go

```diff
@@ -113,12 +113,17 @@ func (c *component) CheckOnce() {
 		return
 	}
 
+	if !netutil.IsPortOpen(c.kubeletReadOnlyPort) {
+		d.err = fmt.Errorf("kubelet is installed but kubelet read-only port %d is not open", c.kubeletReadOnlyPort)
+		return
+	}
+
 	// below are the checks in case "kubelet" is installed, thus requires activeness checks
-	cctx, ccancel := context.WithTimeout(c.ctx, 15*time.Second)
-	running := checkKubeletReadOnlyPortListening(cctx, c.kubeletReadOnlyPort)
+	cctx, ccancel := context.WithTimeout(c.ctx, 30*time.Second)
+	err := checkKubeletReadOnlyPortHealthzWithRetries(cctx, c.kubeletReadOnlyPort)
```

Comment:
IMO even we don't need to call the function, we can just check if the status returned from `listPodsFromKubeletReadOnlyPort` is ok.